internal/beater/beater.go

```
@@ -512,23 +512,24 @@ func (s *Runner) Run(ctx context.Context) error {
 			finalTracerBatchProcessor,
 		}
 
-		closeTracerProcessor = closeTracerFinalBatchProcessor
-
 		tracerProcessor := append(preBatchProcessors, tracerBatchProcessor)
 		tracerServer, err := newTracerServer(s.config, tracerServerListener, s.logger, tracerProcessor, serverParams.Semaphore, serverParams.MeterProvider)
 		if err != nil {
 			return fmt.Errorf("failed to create self-instrumentation server: %w", err)
 		}
-		g.Go(func() error {
+		go func() error {
 			if err := tracerServer.Serve(tracerServerListener); err != http.ErrServerClosed {
 				return err
```

This error is now getting silently ignored, we should probably log it at least.